### PR TITLE
Added Main build github actions workflow

### PR DESCRIPTION
**Pull request by jaydee029**
For issue galasa-dev/projectmanagement#1881

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        
+        - name: Recycle javadocsite application in ArgoCD
+          run: |
+            docker run --env ARGOCD_AUTH_TOKEN=${{ env.ARGOCD_AUTH_TOKEN }} --rm -v ${{ github.workspace }}:/var/workspace ghcr.io/galasa-dev/argocdcli:main app actions run gh-maven-repos restart --kind Deployment --resource-name javadocsite-gh --server argocd.galasa.dev
+        
+        - name: Extract metadata for JavadocMavenrepo image
+          id: metadata
+          uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+          with:
+            images: ${{ env.REGISTRY }}/${{ env.NAMESPACE }}/JavadocMavenrepo
```
Comment (jadecarino) [inline .github/workflows/build.yaml:271]: Can we please change this to `javadoc-maven-artefacts` to match the naming convention for all other Docker images that serve the maven repos

Comment (jadecarino) [reply .github/workflows/build.yaml:271]: Please also update the test image in the PR build to this as well for consistency

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+            push: true
+            tags: ${{ steps.metadata.outputs.tags }}
+            labels: ${{ steps.metadata.outputs.labels }}
+            build-args: |
+              dockerRepository=${{env.REGISTRY}}
+              tag=latest
+            
+        - name: Recycle JavadocMavenrepo application in ArgoCD
+          run: |
+            docker run --env ARGOCD_AUTH_TOKEN=${{ env.ARGOCD_AUTH_TOKEN }} --rm -v ${{ github.workspace }}:/var/workspace ghcr.io/galasa-dev/argocdcli:main app actions run gh-maven-repos restart --kind Deployment --resource-name JavadocMavenrepo-gh --server argocd.galasa.dev
```
Comment (jadecarino) [inline .github/workflows/build.yaml:288]: This should be `--resource-name javadoc-gh` - I apologise I didn't provide you with the names for the resource so you wouldn't have known

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+          uses: docker/build-push-action@v5
+          with:
+            context: .
+            file: ${{github.workspace}}/obr/dockerfiles/dockerfile.javadocmavenrepo
+            push: true
+            tags: ${{ steps.metadata.outputs.tags }}
+            labels: ${{ steps.metadata.outputs.labels }}
+            build-args: |
+              dockerRepository=${{env.REGISTRY}}
+              tag=latest
```
Comment (jadecarino) [inline .github/workflows/build.yaml:284]: This should be `baseVersion=latest` - please also make this same change in the PR workflow

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        
+        - name: Recycle OBR application in ArgoCD
+          run: |
+            docker run --env ARGOCD_AUTH_TOKEN=${{ env.ARGOCD_AUTH_TOKEN }} --rm -v ${{ github.workspace }}:/var/workspace ghcr.io/galasa-dev/argocdcli:main app actions run gh-maven-repos restart --kind Deployment --resource-name obr-gh --server argocd.galasa.dev
+
+    
+  build-obr-javadocs:
+    name: Build OBR javadocs using galasabld image and maven
+    runs-on: ubuntu-latest
+    needs: build-obr
```
Comment (jadecarino) [inline .github/workflows/build.yaml:155]: This can be removed the jobs can run in parallel - please also make the same change in the PR workflow

**Review by jadecarino — CHANGES_REQUESTED** [review]
Good work thanks Dhruv - just a few small changes

File: .github/workflows/build.yaml (hunk 4 above)
Comment (jaydee029) [reply .github/workflows/build.yaml:155]: done

File: .github/workflows/build.yaml (hunk 3 above)
Comment (jaydee029) [reply .github/workflows/build.yaml:284]: done

File: .github/workflows/build.yaml (hunk 2 above)
Comment (jaydee029) [reply .github/workflows/build.yaml:288]: done

File: .github/workflows/build.yaml (hunk 1 above)
Comment (jaydee029) [reply .github/workflows/build.yaml:271]: done

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Review by jadecarino — APPROVED** [review]
Approved for building

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**